File: working/recursividad/21-fib/py/rec.py
class Entero:
    def __init__(self,n):
        self.__n = n

    def fibonacci(self):
        return self.fibonacci_rec(self.__n)

    def fibonacci_rec(self,num):
        if num == 0 or num == 1:
            return num
        else:
            return self.fibonacci_rec(num-1) + self.fibonacci_rec(num-2)


print(Entero(6).fibonacci())

File: working/recursividad/21-fib/rb/rec.rb
class Entero
    def initialize(n)
        @n = n
    end

    def fibonacci
        fibonacci_rec(@n)
    end

    private

    def fibonacci_rec(num)
        if num == 0 || num == 1
            return num
        else
            return fibonacci_rec(num-1) + fibonacci_rec(num-2)
        end
    end
end


puts Entero.new(6).fibonacci
